diff -u ghostscript-8.11/lib/gs_init.ps ghostscript-8.11-easysw/lib/gs_init.ps
--- ghostscript-8.11/lib/gs_init.ps	2003-07-30 03:26:55.000000000 -0400
+++ ghostscript-8.11-easysw/lib/gs_init.ps	2003-11-13 13:04:17.000000000 -0500
@@ -139,6 +139,7 @@
 currentdict /NOFONTPATH known   /NOFONTPATH exch def
 currentdict /NOGC known   /NOGC exch def
 currentdict /NOINTERPOLATE .knownget { /INTERPOLATE exch not def } if
+currentdict /NOMEDIAATTRS known /NOMEDIAATTRS exch def
 currentdict /NOOUTERSAVE known   /NOOUTERSAVE exch def
 currentdict /NOPAGEPROMPT known   /NOPAGEPROMPT exch def
 currentdict /NOPAUSE known   /NOPAUSE exch def
diff -u ghostscript-8.11/lib/gs_setpd.ps ghostscript-8.11-easysw/lib/gs_setpd.ps
--- ghostscript-8.11/lib/gs_setpd.ps	2003-04-12 14:08:18.000000000 -0400
+++ ghostscript-8.11-easysw/lib/gs_setpd.ps	2003-11-14 16:32:55.000000000 -0500
@@ -344,21 +344,42 @@
     } bind
 .dicttomark readonly def
 
-% Define the keys used in input attribute matching.
-/.inputattrkeys [
-  /PageSize /MediaColor /MediaWeight /MediaType /InsertSheet
-	% The following are documented in Adobe's supplement for v2017.
-  /LeadingEdge /MediaClass
-] readonly def
-% Define other keys used in media selection.
-/.inputselectionkeys [
-  /MediaPosition /Orientation
-] readonly def
-
-% Define the keys used in output attribute matching.
-/.outputattrkeys [
-  /OutputType
-] readonly def
+% M. Sweet, Easy Software Products:
+%
+% Define NOMEDIAATTRS to turn off the default (but unimplementable) media
+% selection policies for setpagedevice.  This is used by CUPS to support
+% the standard Adobe media attributes.
+NOMEDIAATTRS {
+  % Define only PageSize for input attribute matching.
+  /.inputattrkeys [
+    /PageSize
+  ] readonly def
+  % Define no other keys used in media selection.
+  /.inputselectionkeys [
+    /noInputSelectionsKeys
+  ] readonly def
+
+  % Define no keys used in output attribute matching.
+  /.outputattrkeys [
+    /noOutputAttrKeys
+  ] readonly def
+} {
+  % Define the keys used in input attribute matching.
+  /.inputattrkeys [
+    /PageSize /MediaColor /MediaWeight /MediaType /InsertSheet
+	  % The following are documented in Adobe's supplement for v2017.
+    /LeadingEdge /MediaClass
+  ] readonly def
+  % Define other keys used in media selection.
+  /.inputselectionkeys [
+    /MediaPosition /Orientation
+  ] readonly def
+
+  % Define the keys used in output attribute matching.
+  /.outputattrkeys [
+    /OutputType
+  ] readonly def
+} ifelse
 
 % Define all the parameters that should always be copied to the merged
 % dictionary.
@@ -387,7 +408,13 @@
   /PageSize //false		% obsolete alias for .MediaSize
   /InputAttributes //false
   .inputattrkeys
-    { dup /PageSize eq
+    % M. Sweet, Easy Software Products:
+    %
+    % Treat LeadingEdge like PageSize so that a common Ghostscript driver
+    % doesn't need the NOMEDIAATTRS definition.
+    %
+    % { dup /PageSize eq
+    { dup dup /PageSize eq exch /LeadingEdge eq or
        { pop }
        { { 2 index /InputAttributes .knownget { //null eq } { //true } ifelse } }
       ifelse
@@ -468,11 +495,22 @@
 % They are expected to consume the top 2 operands.
 % NOTE: we currently treat all values other than 0, 1, or 7 (for PageSize)
 % the same as 0, i.e., we signal an error.
+%
+% M. Sweet, Easy Software Products:
+%
+% Define NOMEDIAATTRS to turn off the default (but unimplementable) media
+% selection policies for setpagedevice.  This is used by CUPS to support
+% the standard Adobe media attributes.
   0 {		% Set errorinfo and signal a configurationerror.
+      NOMEDIAATTRS {
+        % NOMEDIAATTRS means that the default policy is 7...
+        pop 2 index exch 7 put
+      } {
 	pop dup 4 index exch get 2 array astore
 	$error /errorinfo 3 -1 roll put
 	cleartomark
 	/setpagedevice load /configurationerror signalerror
+      } ifelse
   } bind
   1 {		% Roll back the failed request to its previous status.
 SETPDDEBUG { (Rolling back.) = pstack flush } if
